sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -842,7 +856,10 @@ class CodegenContext {
       blocks.head
     } else {
       val func = freshName(funcName)
-      val argString = arguments.map { case (t, name) => s"$t $name" }.mkString(", ")
+      val argString = arguments.map { case (t, name) =>
+        assert(!isDeclaredMutableState(name),
```

Comment:
I'm neutral about this pr though, I feel this is not the best place for this assertion cuz this is not only the place where the suggested case happens, e.g., my pr splits aggregate functions in `HashAggregate`

Reply:
Got it. I did not know #19082 has own split routine. While it is under review, there is no way to add assertion by this PR.

Reply:
IMHO we'd better to check arguments in all the registered functions via `addFunction` if you check the case described in the description?

Reply:
Yeah, this problem occurs not only in split method also in normal method. Let us check this in `addNewFunction`